integration_tests/pages/refer/show/sentenceInformation.ts

```diff
@@ -25,6 +25,17 @@ export default class SentenceInformationPage extends Page {
     )
   }
 
+  shouldContainReleaseDatesSummaryCard(): void {
+    cy.get('[data-testid="release-dates-summary-card"]').then(summaryCardElement => {
+      this.shouldContainSummaryCard(
+        'Release dates',
+        [],
+        PersonUtils.releaseDatesSummaryListRows(this.person),
```

Comment:
I know we've used `utils` like this in a couple of places, which I may have missed at the time they were merged into `main`, but if the implementation/shape of the return value to `PersonUtils.releaseDatesSummaryListRows` changes, this test would still pass - we're essentially testing the implementation, using the implementation here, which doesn't feel very robust.

I'd prefer if we could write out the expected summary list rows here for this as an integration test as it'd catch this if it had an unexpected return value.

Reply:
Is this definitely the approach we want to take (I feel like we might have agreed to take the `server/utils` approach at some point, but not documented it!)? We use utils in a lot of places in the integration tests, as do [Approved Premises](https://github.com/search?q=repo:ministryofjustice/hmpps-approved-premises-ui+path:integration_tests+server/utils&type=code) and [Temporary Accommodation](https://github.com/search?q=repo:ministryofjustice/hmpps-temporary-accommodation-ui%20path:cypress_shared%20server/utils&type=code). The [Approved Premises summary list rows integration test method](https://github.com/search?q=repo:ministryofjustice/hmpps-approved-premises-ui%20shouldcontainsummarylistitems&type=code) is used in a similar way (passing in utils output), and Temporary Accommodation do something similar with [embedded records](https://github.com/search?q=repo:ministryofjustice/hmpps-temporary-accommodation-ui%20assertdefinition&type=code)

I can see why we'd want the integration tests not to be aware of implementation details but we're doing that all over the place, so I think this might be a bit of a substantial refactor - we might need to reimplement some of the utils code that's already unit tested elsewhere to make this dynamic, or set up multiple scenarios for the different earliest release dates so that we can have more hardcoded values (and similar in a few other contexts), which might be a bit overkill here? Maybe better to add this as a backlog ticket if it's important enough?